hoomd/md/methods.py

```diff
@@ -454,69 +454,182 @@ def thermalize_thermostat_and_barostat_dof(self, seed):
+            of isotropic pressure P (:math:`[p, p, p, 0, 0, 0]`), use ``S = p``.
 
-    The available options are identical to those of :py:class:`NPT`, except that *kT* cannot be specified.
-    For further information, refer to the documentation of :py:class:`NPT`.
+        tauS (`float`): Coupling constant for the barostat (in time units).
 
-    Note:
-         A time scale *tauP* for the relaxation of the barostat is required. This is defined as the
-         relaxation time the barostat would have at an average temperature *T_0 = 1*, and it
-         is related to the internally used (Andersen) Barostat mass :math:`W` via
-         :math:`W=d N T_0 \tau_P^2`, where :math:`d` is the dimensionality and :math:`N` the number
-         of particles.
+        couple (`str`): Couplings of diagonal elements of the stress tensor,
+            can be "none", "xy", "xz","yz", or "all", default to "all".
 
-    :py:class:`nph` is an integration method and must be used with ``mode_standard``.
+        box_dof(`tuple` [ `bool` ]): Box degrees of freedom with six boolean
+            elements corresponding to x, y, z, xy, xz, yz, each. Default to
+            [True,True,True,False,False,False]). If turned on to True,
+            rescale corresponding lengths or tilt factors and components of
+            particle coordinates and velocities.
+
+        rescale_all (`bool`): if True, rescale all particles, not only those in
+            the group, Default to False.
+
+        gamma (`float`): Dimensionless damping factor for the box degrees of
+            freedom, Default to 0.
+
+    Note:
+        Coupling constant for barostat `tauS` should be set within appropriate
+        range for pressure and volume to fluctuate in reasonable rate and
+        equilibrate. Too small `tauS` can cause abrupt fluctuation, whereas too
+        large `tauS` would take long time to equilibrate. In most of systems,
+        recommended value for `tauS` is ``1000 * dt``, where ``dt`` is the
+        length of the time step.
 
     Examples::
+        dt = 0.005
+        tauS = 1000 * dt
+        nph = hoomd.md.methods.NPH(filter=hoomd.filter.All(), tauS=tauS, S=2.0)
+        # orthorhombic symmetry
+        nph = hoomd.md.methods.NPH(filter=hoomd.filter.All(), tauS=tauS, S=2.0, couple="none")
+        # tetragonal symmetry
+        nph = hoomd.md.methods.NPH(filter=hoomd.filter.All(), tauS=tauS, S=2.0, couple="xy")
+        # triclinic symmetry
+        nph = hoomd.md.methods.NPH(filter=hoomd.filter.All(), tauS=tauS, S=2.0, couple="none", rescale_all=True)
+        integrator = hoomd.md.Integrator(dt=dt, methods=[nph], forces=[lj])
+
+    Attributes:
+        filter (hoomd.filter.ParticleFilter): Subset of particles on which to
+            apply this method.
+
+        S (tuple[hoomd.variant.Variant, hoomd.variant.Variant, hoomd.variant.Variant, hoomd.variant.Variant, hoomd.variant.Variant, hoomd.variant.Variant]): Stress components set
```

Comment:
```suggestion
        S (tuple[hoomd.variant.Variant, ...]): Stress components set
```
While not fully correct as we expect exactly 6 elements, this is much more concise and follows the type hints guide losely,  https://stackoverflow.com/questions/47533787/typehinting-tuples-in-python.